src/wallet/wallet.h

```diff
@@ -859,6 +862,8 @@ class CWallet final : public CCryptoKeyStore, private interfaces::Chain::Notific
     void AbortRescan() { fAbortRescan = true; }
     bool IsAbortingRescan() { return fAbortRescan; }
     bool IsScanning() { return fScanningWallet; }
+    int64_t ScanningDuration() const { return fScanningWallet ? GetTimeMillis() - m_scanning_start : 0; }
```

Comment:
shouldn't this (in theory) lock `mutexScanning` to avoid the risk of an out-of-sync `fScanningWallet`/`m_scanning_start` combination (same for line below and wallet.cpp:L1795)?

Reply:
`m_scanning_start` is constant right before `fScanningWallet` is set to `true` so in this case I don't think it's necessary.

The atomic `m_scanning_progress` is only read when `fScanningWallet` is `true`. Also, `m_scanning_progress` is only changed in `ScanForWalletTransactions`, and we ensure there's only one scan in progress.